docs/data/base/guides/interoperability-with-tailwind-css/interoperability-with-tailwind-css.md

```diff
@@ -0,0 +1,451 @@
+This is because we haven't defined the element that represents the dot inside the thumb.
+In order to do this, is not enough to just propagate classes for the thumb, we need to actually render a custom component for it.
+We can do this easily by passing the custom component in the `components` props of the `Slider`.
+This is what we need to do
+
+```diff
+--- a/src/Slider.tsx
++++ b/src/Slider.tsx
+@@ -1,6 +1,17 @@
+ import * as React from 'react';
+ import SliderUnstyled, { SliderUnstyledThumbSlotProps, SliderUnstyledProps } from '@mui/base/SliderUnstyled';
+
++const Thumb = React.forwardRef(function Thumb(
++  props: SliderUnstyledThumbSlotProps,
++  ref: React.ForwardedRef<HTMLSpanElement>,
++) {
++  const { ownerState, className, children, ...other } = props;
++  return (<span className={`${className} ring-cyan-500 dark:ring-cyan-400 ring-2 w-4 h-4 -mt-1 -ml-2 flex items-center justify-center bg-white rounded-full shadow absolute`} {...other} ref={ref}>
++    <span className="w-1.5 h-1.5 bg-cyan-500 dark:bg-cyan-400 rounded-full ring-1 ring-inset ring-slate-900/5"></span>
++    {children}
++  </span>);
++});
++
+ const Slider = React.forwardRef(function Slider(
+   props: SliderUnstyledProps,
+   ref: React.ForwardedRef<HTMLSpanElement>,
+@@ -8,9 +19,11 @@ const Slider = React.forwardRef(function Slider(
+   return (<SliderUnstyled
+     {...props}
+     ref={ref}
++    components={{
++      Thumb,
++    }}
+     componentsProps={{
+       root: { className: "w-full relative inline-block h-2 cursor-pointer" },
+-      thumb: { className: "ring-cyan-500 dark:ring-cyan-400 ring-2 w-4 h-4 -mt-1 -ml-2 flex items-center justify-center bg-white rounded-full shadow absolute" },
+       rail: { className: "bg-slate-100 dark:bg-slate-700 h-2 w-full rounded-full block absolute" },
+       track: { className: "bg-cyan-500 dark:bg-cyan-400 h-2 absolute rounded-full" }
+     }}
+```
+
+If you refresh the page now, you will see that the thumb looks idential as in the first picture.
+All that we did is, created custom component, appended the same classes as we did before on the root element, and before rendering the default children provided by the Slider component, we've added a span that represents the dot inside the thumb.
+This is just a small example, but you can image building much more custom component for each of the slots available in the MUI Base components.
+
+:::warning
+⚠️ When you are building building custom components for the slots, it is important to always <b>propagate the props sent from the owner component on the root element</b>.
+This is important, as the props will contain event handlers and aria properties necessaryf or making the component accessible.\
+In addition to this, each of the slots receives an `ownerState` object, which contains the props & state of the owner component. This is useful if you want to style the component based on some internal state.
+:::
+
+## Adding custom focus visible selector on the buttons
```

Comment:
```suggestion
## Adding a custom `focus-visible` selector on the buttons
```

Reply:
it's misleading if we use `focus-visible` here as it is valid pseudo selector. We are rather using `focusVisible` state which is in a way polyfill for it.

Reply:
```suggestion
## Adding a custom focus visible selector on the buttons
```